checker/src/summaries.rs

```diff
@@ -210,6 +210,20 @@ impl<'a, 'tcx: 'a> PersistentSummaryCache<'a, 'tcx> {
     /// Any summaries that are still in a default state will be for functions in the foreign_contracts
     /// module and their definitions use DefIds that are different from the ids at the call sites.
     pub fn invalidate_default_summaries(&mut self) {
+        let keys_to_remove: Vec<usize> = self
```

Comment:
It'd be nice if the caches had an 'invalidate_default_summaries' method you could call, or at least a generic that implements the logic for both. I had to play 'spot the difference' again.